catkin_ws/src/stability/src/AxialPid.cpp

```diff
@@ -0,0 +1,257 @@
+#include "AxialPid.h"
+#include <geometry_msgs/Vector3.h>
+#include <sensor_msgs/Imu.h>
+#include <ros/ros.h>
+#include <cmath>
+
+RotPIDController::RotPIDController(float newRKp, float newRKi, float newRKd, float newPKp, float newPKi, float newPKd, float newYKp, float newYKi, float newYKd, float rTarget, float pTarget, float yTarget) :
+	//initializing PID constants & targets
+	rollKp(newRKp), rollKi(newRKi), rollKd(newRKd), pitchKp(newPKp), pitchKi(newPKi), pitchKd(newPKd), yawKp(newYKp), yawKi(newYKi), yawKd(newYKd), rollTarget(rTarget), pitchTarget(pTarget), yawTarget(yTarget),
+	//initializing PID controllers
+	rollControl(rollKp, rollKi, rollKd), pitchControl(pitchKp, pitchKi, pitchKd), yawControl(yawKp, yawKi, yawKd)
+{
+	OUT_LIMIT = 1;
+	MARGIN = 0.5;
+	//setting output limits
+	rollControl.setOutputLimits(OUT_LIMIT);
+	pitchControl.setOutputLimits(OUT_LIMIT);
+	yawControl.setOutputLimits(OUT_LIMIT);	
+	//setting pid targets
+	rollControl.setSetpoint(0);
+	pitchControl.setSetpoint(0);
+	yawControl.setSetpoint(0);
+	rollFlip = pitchFlip = yawFlip = false;
+
+}
+
+RotPIDController::RotPIDController(float newRKp, float newRKi, float newRKd, float newPKp, float newPKi, float newPKd, float newYKp, float newYKi, float newYKd) :
+	//initializing PID constants & targets
+	rollKp(newRKp), rollKi(newRKi), rollKd(newRKd), pitchKp(newPKp), pitchKi(newPKi), pitchKd(newPKd), yawKp(newYKp), yawKi(newYKi), yawKd(newYKd), rollTarget(0), pitchTarget(0), yawTarget(0),
+	//setting PID controllers
+	rollControl(rollKp, rollKi, rollKd), pitchControl(pitchKp, pitchKi, pitchKd), yawControl(yawKp, yawKi, yawKd)
+{
+	OUT_LIMIT = 1;
+	MARGIN = 0.5;
+	//setting pid limits
+	rollControl.setOutputLimits(OUT_LIMIT);
+	pitchControl.setOutputLimits(OUT_LIMIT);
+	yawControl.setOutputLimits(OUT_LIMIT);
+
+	//setting pid targets
+	rollControl.setSetpoint(0);
+	pitchControl.setSetpoint(0);
+	yawControl.setSetpoint(0);
+
+	rollFlip = pitchFlip = yawFlip = false;
+}
+
+RotPIDController::RotPIDController():
+	//initializing PID constants and targets
+	rollKp(0), rollKi(0), rollKd(0), pitchKp(0), pitchKi(0), pitchKd(0), yawKp(0), yawKi(0), yawKd(0), rollTarget(0), pitchTarget(0), yawTarget(0),
+	//initializing PID controllers
+	rollControl(rollKp, rollKi, rollKd), pitchControl(pitchKp, pitchKi, pitchKd), yawControl(yawKp, yawKi, yawKd)
+{
+	OUT_LIMIT = 1;
+	MARGIN = 0.5;
+	//setting pid limits
+	rollControl.setOutputLimits(OUT_LIMIT);
+	pitchControl.setOutputLimits(OUT_LIMIT);
+	yawControl.setOutputLimits(OUT_LIMIT);
+
+	//setting pid targets
+	rollControl.setSetpoint(0);
+	pitchControl.setSetpoint(0);
+	yawControl.setSetpoint(0);
+
+	rollFlip = pitchFlip = yawFlip = false;
+	std::cout<<"a"<<std::endl;
+}
+
+	
+//returns all pid outputs in a vector in this order (roll, pitch, yaw)
+void RotPIDController::getMotorValues(float rollValue, float pitchValue, float yawValue, float& rollOut, float& pitchOut, float& yawOut)

```

Comment:
minor comment, but a better name might be getCommandValues, as these are not directly connected to any motor values

Reply:
ok did the changes